tests/workflows/await_any_conditional_loops/workflow.py

```diff
@@ -0,0 +1,88 @@
+import time
+
+from vellum.workflows.nodes.bases.base import BaseNode
+from vellum.workflows.ports.port import Port
+from vellum.workflows.state.base import BaseState
+from vellum.workflows.types.core import MergeBehavior
+from vellum.workflows.workflows.base import BaseWorkflow
+
+
+class StartNode(BaseNode):
+    pass
+
+
+class TopNode(BaseNode):
+    class Ports(BaseNode.Ports):
+        success = Port.on_if(StartNode.Execution.count.equals(1))
+        failure = Port.on_else()
+
+
+class TopDeadEndNode(BaseNode):
+    pass
+
+
+class BottomNode(BaseNode):
+    class Ports(BaseNode.Ports):
+        success = Port.on_if(StartNode.Execution.count.equals(2))
+        failure = Port.on_else()
+
+
+class BottomDeadEndNode(BaseNode):
+    pass
+
+
+class MergeNode(BaseNode):
+    class Trigger(BaseNode.Trigger):
+        merge_behavior = MergeBehavior.AWAIT_ANY
+
+
+class LoopNode(BaseNode):
+    class Ports(BaseNode.Ports):
+        loop = Port.on_if(MergeNode.Execution.count.equals(1))
+        exit = Port.on_else()
+
+
+class ExitNode(BaseNode):
+    class Outputs(BaseNode.Outputs):
+        final_value = LoopNode.Execution.count
+
+
+class AwaitAnyWithConditionalLoopsWorkflow(BaseWorkflow):
+    r"""
+    This is a trickier example of the AWAIT_ANY merge behavior.
+
+    Here's the Ascii diagram:
+
+                     TopDeadEndNode
+                     /
+              TopNode                       StartNode
```

Comment:
algorithm for auto-detecting this loop:
- Traverse the graph from StartNode
- When you detect a Loop who's target is in set of discovered nodes, automatic fork
    - "basically DFS"